qa/rpc-tests/test_framework/util.py

```diff
@@ -279,15 +279,16 @@ def is_connected(gdict, vertices_encountered = None, start_vertex=None):
             return True
         return False
 
-def sync_blocks(rpc_connections, wait=1,verbose=1):
+def sync_blocks(rpc_connections, *, wait=1, verbose=1, timeout=60):
     """
     Wait until everybody has the same block count
     """
     iter=-1
-    while True:
+    stop_time = time.time() + timeout
+    while time.time() <= stop_time:
         counts = [ x.getblockcount() for x in rpc_connections ]
         if counts == [ counts[0] ]*len(counts):
-            break
+            return
```

Comment:
shouldn't you take out the return and leave it as "break"?

Reply:
mmm the break will bring the code of out of the while loop and then we are going to raise the timeout exception  even in the case the timeout won't get reached, no?

Reply:
duh, yes right...